src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java

```diff
@@ -696,14 +819,803 @@ public boolean validatePortRange(Integer portNumber) {
+					if (value.equals(cachedStreamConfig.getName()) || value.equals(cachedStreamConfig.getDefaultStreamName())) {
+						primaryStreamID = cachedStreamConfig.getId();
+						primaryStreamName = value;
+						break;
+					}
+				}
+				cachedDecoderConfig.setPrimaryStream(primaryStreamID);
+				this.cachedDecoderConfigs.set(decoderID, cachedDecoderConfig);
+				addAdvanceControlProperties(advancedControllableProperties, createDropdown(stats, decoderControllingGroup + DecoderControllingMetric.PRIMARY_STREAM.getName(), streamNames, primaryStreamName));
+				populateApplyChangeAndCancelButtonForDecoder(stats, advancedControllableProperties, decoderID);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case SECONDARY_STREAM:
+				String secondaryStreamID = DecoderConstant.DEFAULT_STREAM_ID;
+				String secondaryStreamName = DecoderConstant.NONE;
+				for (StreamConfig cachedStreamConfig : cachedStreamConfigs) {
+					if (value.equals(cachedStreamConfig.getName()) || value.equals(cachedStreamConfig.getDefaultStreamName())) {
+						secondaryStreamID = cachedStreamConfig.getId();
+						secondaryStreamName = value;
+						break;
+					}
+				}
+				cachedDecoderConfig.setPrimaryStream(secondaryStreamID);
+				this.cachedDecoderConfigs.set(decoderID, cachedDecoderConfig);
+				addAdvanceControlProperties(advancedControllableProperties,
+						createDropdown(stats, decoderControllingGroup + DecoderControllingMetric.PRIMARY_STREAM.getName(), streamNames, secondaryStreamName));
+				populateApplyChangeAndCancelButtonForDecoder(stats, advancedControllableProperties, decoderID);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case STILL_IMAGE:
+				StillImage stillImage = StillImage.getByAPIName(getDefaultValueForNullData(cachedDecoderConfig.getStillImage(), DecoderConstant.EMPTY));
+				cachedDecoderConfig.setStillImage(stillImage.getApiName());
+				this.cachedDecoderConfigs.set(decoderID, cachedDecoderConfig);
+				addAdvanceControlProperties(advancedControllableProperties,
+						createDropdown(stats, decoderControllingGroup + DecoderControllingMetric.STILL_IMAGE.getName(), stillImages, stillImage.getUiName()));
+				populateApplyChangeAndCancelButtonForDecoder(stats, advancedControllableProperties, decoderID);
+				populateLocalExtendedStats(stats, advancedControllableProperties);
+				break;
+			case STILL_IMAGE_DELAY:
+				Integer stillImageDelay = DecoderConstant.DEFAULT_STILL_IMAGE_DELAY;
+				try {
+					stillImageDelay = Integer.parseInt(value);
+					if (stillImageDelay < DecoderConstant.MIN_STILL_IMAGE_DELAY) {
+						stillImageDelay = DecoderConstant.MIN_STILL_IMAGE_DELAY;
+					}
+					if (stillImageDelay > DecoderConstant.MAX_STILL_IMAGE_DELAY) {
+						stillImageDelay = DecoderConstant.MAX_STILL_IMAGE_DELAY;
+					}
+				} catch (Exception e) {
+					if (logger.isWarnEnabled()) {
+						logger.warn("Invalid still image delay value");
```

Comment:
Also log for catched exception.
logger.warn("Invalid still image delay value", e);

Reply:
I fixed it